### PR TITLE
fix: await pending transactions before changing workspace details

### PR DESCRIPTION
**Pull request by nsrCodes**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

**Comment by sagarsoni7** [comment]
Thanks Navdeep, will complete review by tonight.

**Review by sagarsoni7 — CHANGES_REQUESTED** [review]

File: app/src/utils/StorageServiceWrapper.js [hunk 1]
```diff
@@ -17,6 +17,27 @@ class StorageServiceWrapper {
     this.saveRecord = this.saveRecord.bind(this);
     this.getRecord = this.getRecord.bind(this);
     this.getRecords = this.getRecords.bind(this);
+
+    this.transactionLedger = {};
```
[the remark this answers is not in the document]
Comment (sagarsoni7) [reply app/src/utils/StorageServiceWrapper.js:21]: https://github.com/requestly/requestly/pull/1420

[the remark this answers is not in the document]
Comment (nsrCodes) [reply app/src/utils/StorageServiceWrapper.js]: thanks for the improvements @sagarsoni7

File: app/src/utils/StorageServiceWrapper.js [hunk 2]
```diff
@@ -17,6 +17,27 @@ class StorageServiceWrapper {
...
+    this.transactionQueue = []; // promises of transactions that are still pending
+  }
+  // TODO: CLEANUP
+  trackPromise(promise) {
+    const id = Math.random().toString(36).substring(7);
+    console.log("promise id", id);
+    const index = this.transactionQueue.push(promise) - 1;
+    this.transactionLedger[id] = index;
+    promise.then(() => {
```
[the remark this answers is not in the document]
Comment (nsrCodes) [reply app/src/utils/StorageServiceWrapper.js]: also covered in https://github.com/requestly/requestly/pull/1420

**Review by sagarsoni7 — APPROVED** [review]

**Pull request merged**